### PR TITLE
pymupdf: add patches for nix

### PR DESCRIPTION
**Pull request by luhuaei**
###### Motivation for this change
#126384

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133211` at da0c0352 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pymupdf</li>
    <li>python39Packages.pymupdf</li>
    <li>termpdfpy</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/pymupdf/default.nix:15:5:
```
   |
15 |     ./nix-support.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pymupdf/default.nix:33:5:
```
   |
33 |     license = licenses.agpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pymupdf/default.nix [hunk 1]
```diff
@@ -1,28 +1,53 @@
...
 
   src = fetchPypi {
     pname = "PyMuPDF";
     inherit version;
-    sha256 = "64ce58f92d9edd2631e447175fa13b4024ac3b6dce6e718e0b003c41de6f7952";
+    sha256 = "b21e39098fbbe0fdf269fdb2d1dd25a3847bbf22785ee8903d3a5637c2d0b9d7";
   };
 
-  patchPhase = ''
+  patchFlags = [ "--binary" "--ignore-whitespace" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymupdf/default.nix:25]: Why are those flags necessary especially binary?

File: pkgs/development/python-modules/pymupdf/default.nix [hunk 2]
```diff
@@ -1,28 +1,53 @@
...
+  patches = [
+    # Add NIX environment support.
+    # Should be removed next pyMuPDF release.
+    (fetchpatch {
+      url = "https://github.com/pymupdf/PyMuPDF/commit/d9b2d42019e5705a1c6621ea0cdfa26da1ce9ad5.patch";
+      sha256 = "fc3f6ad88c8f3933ed9ab9d4db9ebec8bc30ed5113f6ca9d72080b56dfa52ad6";
+    })
+  ];
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymupdf/default.nix:35]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/pymupdf/default.nix [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
+  buildInputs = [ mupdf freetype harfbuzz openjpeg jbig2dec libjpeg_turbo gumbo ];
 
   doCheck = false;
 
+  pythonImportsCheck = [ "fitz" ];
+
   meta = with lib; {
     description = "Python bindings for MuPDF's rendering library.";
     homepage = "https://github.com/pymupdf/PyMuPDF";
     maintainers = with maintainers; [ teto ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymupdf/default.nix:49]: Do you want to add yourself as a maintainer?

File: pkgs/development/python-modules/pymupdf/default.nix (hunk 1 above)
Comment (luhuaei) [reply pkgs/development/python-modules/pymupdf/default.nix:25]: I local test found the `setup.py` is DOS type, but the patch file is `UTF-8`, so the `patch` command will failed.

File: pkgs/development/python-modules/pymupdf/default.nix (hunk 3 above)
Comment (luhuaei) [reply pkgs/development/python-modules/pymupdf/default.nix:49]: OK :smile:

File: pkgs/development/python-modules/pymupdf/default.nix (hunk 2 above)
Comment (luhuaei) [reply pkgs/development/python-modules/pymupdf/default.nix]: Thank suggest, already changed.

File: pkgs/development/python-modules/pymupdf/default.nix [hunk 4]
```diff
@@ -1,28 +1,53 @@
...
   doCheck = false;
 
+  pythonImportsCheck = [ "fitz" ];
+
   meta = with lib; {
     description = "Python bindings for MuPDF's rendering library.";
     homepage = "https://github.com/pymupdf/PyMuPDF";
-    maintainers = with maintainers; [ teto ];
-    license =  licenses.agpl3;
+    maintainers = with maintainers; [ teto luhuaei ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymupdf/default.nix:49]: Ah, it is your first PR. I didn't look at that. You can drop that change again if you don't want to add yourself to the maintainer list.

**Pull request merged**